### ISSUE TITLE
/home/user is empty

### ISSUE THREAD

**Issue by ghost** [pymumu/jailed-shell#16]
I jailed an already created user using your commands.
The /home/user directory has files and directories like public_html, mail, www, ssl, etc.
But in the jailed environment, the /home/user directory is empty.
I checked that in WinSCP. 
I want that when I jail user, the home folder should retain all its files and folders.

**Comment by pymumu** [comment]
Try the following configuration:

```shell 
bind /home/%u /home/ rw,nodev,nosuid
```

**Comment by jaimbh** [comment]
I tried:
bind /home/%u /home/ rw,nodev,nosuid,
bind /home/user /home/ rw,nodev,nosuid, and
bind /home/user /home/user rw,nodev,nosuid

They all gave me this:
readline: /home/user: no key sequence terminator
readline: /home/: no key sequence terminator
readline: rw,nodev,nosuid: no key sequence terminator

**Comment by jaimbh** [comment]
I tried adding bind /home/%u /home/ rw,nodev,nosuid in configuration.
I even tried bind /home/username /home/ rw,nodev,nosuid and bind /home/username /home/username rw,nodev,nosuid.
It did not work.
The /home/username folder still shows empty after jailing the user.

**Comment by jaimbh** [comment]
I also tried creating symlink for /home/user in the jail. But it gave me error: "Too many levels of symbolic links".
I tried using clink too. It did not work well.

**Comment by ghost** [comment]
yes it worked after I added your given command in the jail configuration below:
#Base directory binding configuration
#Set directory read-only, and prohibit device files 
bind / ro,nodev,nosuid
bind /dev ro,dev,noexec,nosuid
bind /home/user /home/user rw,nodev,nosuid

But when I try to open a file inside /home/user through winSCP, I get the below error:
Cannot execute SCP to start transfer. Please make sure that SCP is installed on the server and path to it is included in PATH. You may also try SFTP instead of SCP.
Command failed with return code 127.

I got this error when I tried opening some files in /home/user.
Thanks

**Comment by pymumu** [comment]
You should copy scp command into jail.
```
clink  /usr/bin/scp /usr/bin/scp
```

**Comment by jaimbh** [comment]
yes it worked thanks.

I want the entire /var folder in the jail along with the files and folders that are inside /var.

I tried adding the below command in the jail config:
bind /var / rw,nodev,nosuid

It did not work.

I placed the command below in the jail config like this:
#Base directory binding configuration
#Set directory read-only, and prohibit device files 
bind / ro,nodev,nosuid
bind /dev ro,dev,noexec,nosuid
bind /home/user /home/user rw,nodev,nosuid
bind /var / rw,nodev,nosuid

Is this command wrong or am I placing it in the wrong place?

**Comment by ghost** [comment]
I want to put 5 users in the same jail. But those users should not be able to access other users' /home directory.

**Comment by pymumu** [comment]
```
bind /home/%u /home/ rw,nodev,nosuid
```
%u means user name

**Comment by jaimbh** [comment]
Possible bug:-
I tried removing a user's jail from terminal using below command:
jail-shell jail -r user-jail

It gave the below error:
/usr/local/jail-shell/bin/jail-shell-setup: line 825: remove: command not found

I think remove_jail function should be called in line 825 instead of just remove. What do you think?
![image](https://user-images.githubusercontent.com/115988572/205230783-b0e9c5b7-0334-49da-a1bf-15c863750a67.png)

**Comment by jaimbh** [comment]
Installing test jail in ubuntu20 gives below error:
cp: cannot create symbolic link '/home/jails/test-jail//usr/lib/x86_64-linux-gnu/libnss_compat.so.2': No such file or directory
command failed at line 140: clink /lib/x86_64-linux-gnu/libnss_compat.so.2  /lib/x86_64-linux-gnu/libnss_compat.so.2
setup jail test-jail failed.

I faced this issue only in ubuntu20.
I did not face this issue in ubuntu18, centos 7,8 or almalinux 8.

**Comment by ghost** [comment]
How to give a jailed user access to entire /usr directory using bind mounts? I tried adding "bind /usr /usr ro,nodev,exec,nosuid" in jail conf. It did not work for me.

**Comment by pymumu** [comment]
Bind the entire /usr directory is not supported because the /usr/bin/init program in the jail is required.
